vb-core/src/main/java/scripts/AlKharidSmelter.java

```diff
@@ -256,6 +256,12 @@ private ScriptParams(int barId, Ore primaryOre, Ore secondaryOre) {
             this.primaryOre = primaryOre;
             this.secondaryOre = secondaryOre;
         }
+
+        private ScriptParams(int barId, Ore primaryOre) {
+            this.barId = barId;
+            this.primaryOre = primaryOre;
+            this.secondaryOre = null;
```

Comment:
This null assignment is unnecessary because it's already null (as the default value) before the constructor is invoked. This can be removed.